pkgs/development/mobile/androidenv/examples/shell.nix

```
@@ -0,0 +1,107 @@
+{
+  pkgs ? import ../../../../.. {},
+  pkgs_i686 ? import ../../../../.. { system = "i686-linux"; },
+  config ? pkgs.config
+}:
+
+let
+  android = {
+    versions = {
+      tools = "26.1.1";
+      platformTools = "30.0.5";
+      buildTools = "30.0.2";
+      ndk = "21.3.6528147";
+      cmake = "3.10.2";
+    };
+
+    platforms = ["23" "24" "25" "26" "27" "28" "29"];
+    abis = ["armeabi-v7a" "arm64-v8a"];
+    extras = ["extras;google;gcm"];
+  };
+
+  androidEnv = import ./.. {
+    inherit config pkgs pkgs_i686;
+    licenseAccepted = true;
+  };
+
+  androidComposition = androidEnv.composeAndroidPackages {
+    toolsVersion = android.versions.tools;
+    platformToolsVersion = android.versions.platformTools;
+    buildToolsVersions = [android.versions.buildTools];
+    platformVersions = android.platforms;
+    abiVersions = android.abis;
+
+    includeSources = false;
+    includeSystemImages = false;
+    includeEmulator = false;
+
+    includeNDK = true;
+    ndkVersion = android.versions.ndk;
+    cmakeVersions = [android.versions.cmake];
+
+    useGoogleAPIs = true;
+    includeExtras = android.extras;
+
+    # If you want to use a custom repo JSON:
+    repoJson = ../repo.json;
```

Let's make the file opt-in:

```suggestion
    #repoJson = ./repo.json;
```